src/freenet/node/PeerNode.java

```diff
@@ -601,6 +602,9 @@ public PeerNode(SimpleFieldSet fs, Node node2, NodeCrypto crypto, boolean fromLo
 			Logger.normal(this, "No IP addresses found for identity '" + identityAsBase64String + "', possibly at location '" + location + ": " + userToString());
 			detectedPeer = null;
 		} else {
+			nominalPeer.sort(Peer.PEER_COMPARATOR);
+			// TODO	this throws away all valid addresses but the first, without checking whether they can connect. Need to try a later one if connection fails.
```

Comment:
Is my understanding correct that this isn't a change to behavior, just a note about it?

Reply:
Not completely: this sorts the addresses, so it gets the best address instead of the first. Usually that will be the hostname.